src/main/java/org/inferred/freebuilder/processor/DefaultPropertyFactory.java

```diff
@@ -169,18 +169,56 @@ public void addFinalFieldAssignment(SourceBuilder code, String finalField, Strin
 
     @Override
     public void addMergeFromValue(Block code, String value) {
-      code.addLine("%s(%s.%s());", setter(property), value, property.getGetterName());
+      Excerpt defaults = Declarations.freshBuilder(code, metadata).orNull();
+      if (defaults != null) {
+        code.add("if (");
+        if (!hasDefault) {
+          code.add("%s._unsetProperties.contains(%s.%s) || ",
```

Comment:
should this be something like `if the default value of this property is different from the default, set it` (if not, because we want everything in a value (but not in a template builder) as overriding, I think the `unsetProperties.contains` check on the defaults may be redundant, since my understanding of this is `always set the property if it's something that is not set by default by the Builder's constructor (aka, it's unset), otherwise only set it if the value's property value  is different than this builder's value for the property` 

Couldn't we just have `set this if the property's value is different than this builder's value for the property`?

Also, is this the expected behaviour:

`interface A { Optional<String> getA() }`
`A a = A.builder().build()`
`A b = A.builder().setA("A!").mergeFrom(a).build();`

`b.getA().isPresent()` is `false`?